source/agora/consensus/protocol/Nominator.d

```diff
@@ -649,9 +649,15 @@ extern(D):
             Transaction[] received_tx_set;
             if (auto fail_reason = this.ledger.getValidTXSet(con_data, received_tx_set, this.ledger.getUTXOFinder()))
             {
-                log.info("Missing TXs while checking envelope signature : {}",
-                    scpPrettify(&envelope, &this.getQSet));
-                return; // We dont have all the TXs for this block. Try to catchup
+                if (fail_reason == this.ledger.InvalidConsensusDataReason.NotInPool)
+                {
+                    log.trace("Tx set being confirmed is not yet valid: {}", fail_reason);
+                    this.queued_envelopes.insertBack(envelope.clone()); // Put back in queue
```

Comment:
What about `insertFront`?

Reply:
Why? That would delay processing of other envelopes until we are done with that one and that is not something we want

Reply:
Code has been removed.